aten/src/ATen/ParallelOpenMP.h

```diff
@@ -25,16 +25,17 @@ inline void parallel_for(
 #ifdef _OPENMP
   std::atomic_flag err_flag = ATOMIC_FLAG_INIT;
   std::exception_ptr eptr;
-
-#pragma omp parallel if (!omp_in_parallel() && ((end - begin) > grain_size))
+  // choose number of tasks based on grain size and number of threads
+  // can't use num_threads clause due to bugs in GOMP's thread pool (See #32008)
+  int64_t num_threads = omp_get_num_threads();
```

Comment:
This doesn't work. `omp_get_num_threads` returns the number of threads in the _current_ openmp team. `omp parallel` creates a new team with a different number of threads.

Reply:
What if I revert this, but still call `orig_num_threads = omp_get_num_threads()` once before the pragma so we can avoid entering the `parallel for` when `orig_num_threads` is 1 ?

Reply:
Outside of a parallel region, `omp_get_num_threads()` always returns 1.

Reply:
Perhaps add a `thread_local bool in_parallel_region` ?

Reply:
I used `omp_get_max_threads()`